src/utils.c

```diff
@@ -76,9 +76,9 @@ char *string_replace_all(const char *needle, const char *replacement,
 
 char *string_append(char *a, const char *b, const char *sep)
 {
-        if (!a)
+        if (!a || *a == '\0')
```

Comment:
`a` isn't freed in this case if it's `\0` which could cause an (albeit minor) memory leak.

I have no clue *why* this function needs to free only `a` but that's how it works it's best to preserve that until we can figure out why.

Reply:
> a isn't freed in this case if it's \0 which could cause an (albeit minor) memory leak.

a very good catch!

Reply:
> I have no clue why this function needs to free only a but that's how it works it's best to preserve that until we can figure out why.

It's cool to have such a function to concatenate some stuff to something existing without having to check for anything:

```
n->msg = string_append(n->msg, addition, "\n");
```

In one line, you appended it if it's a non-empty and not-null string.

Reply:
> a isn't freed in this case if it's \0 which could cause an (albeit minor) memory leak.

Now after freeing `a`, I found a missing `g_strdup` for the first parameter in a unittest. Thanks for that good catch.